test/local/push_tests.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var tap = require('tap')
+var test = tap.test
+var extend = require('util')._extend
+
+var P = require('../../lib/promise')
+var mockUid = new Buffer('foo')
+var mockLog = {
+  error: function () {},
+  increment: function () {},
+  trace: function () {}
+}
+
+var mockDbEmpty = {
+  devices: function () {
+    return P.resolve([])
+  }
+}
+
+var mockDbResult = {
+  devices: function (uid) {
+    return P.resolve([{
+      'id': '0f7aa00356e5416e82b3bef7bc409eef',
+      'isCurrentDevice': true,
+      'lastAccessTime': 1449235471335,
+      'name': 'My Phone',
+      'type': 'mobile',
+      'pushCallback': 'https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef',
+      'pushPublicKey': '468601214f60f4828b6cd5d51d9d99d212e7c73657978955f0f5a5b7e2fa1370'
+    }])
+  }
+}
+
+test(
+  'notifyUpdate does not throw on empty device result',
+  function (t) {
+    try {
+      var push = require('../../lib/push')(mockLog, mockDbEmpty)
+      push.notifyUpdate(mockUid)
+    } catch (e) {
+      t.fail('must not throw')
```

Comment:
Should this be an assertion that the `push.notifyUpdate` didn't reject its promise? Or failing that, should we assert both?